lesson_16/txt_proxy_reader_writer.py

```diff
@@ -0,0 +1,49 @@
+from lesson_16.txt_reader import TxtReader
+from lesson_16.txt_writer import TxtWriter
+
+
+class TxtProxyReaderWriter:
+    def __init__(self, file_path):
+        self.__result = ''
+        self.__is_actual = False
+        self.__txt_reader = TxtReader(file_path)
+        self.__txt_writer = TxtWriter(file_path)
+
+    @property
+    def get_result(self):
+        if self.__result == '':
+            return f'No cache'
+        else:
+            return f'DATA IN CACHED FILE: \n {self.__result} \n'
+
+    def read_file(self):
+        if self.__is_actual:
+            return self.__result
+        else:
+            self.__result = self.__txt_reader.read()
+            self.__is_actual = True
+            return self.__result
+
+    def write_file(self, new_data, mode='w'):
```

Comment:
also add check : 
if new_data = self.__result and mode w:

in such case, you do not need to write anything